modules/basegl/src/processors/imageprocessing/imagechannelcombine.cpp

```diff
@@ -0,0 +1,100 @@
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#include <modules/basegl/processors/imageprocessing/imagechannelcombine.h>
+#include <modules/opengl/texture/textureunit.h>
+#include <modules/opengl/texture/textureutils.h>
+#include <modules/opengl/shader/shaderutils.h>
+
+namespace inviwo {
+
+    const ProcessorInfo ImageChannelCombine::processorInfo_{
+        "org.inviwo.ImageChannelCombine",  // Class identifier
+        "Image Channel Combine",           // Display name
+        "Image Operation",                 // Category
+        CodeState::Experimental,           // Code state
+        Tags::GL,                          // Tags
+    };
+    const ProcessorInfo ImageChannelCombine::getProcessorInfo() const { return processorInfo_; }
+
+    ImageChannelCombine::ImageChannelCombine() :
+        Processor()
+        , inport0_("inport0", true)
+        , inport1_("inport1", true)
+        , inport2_("inport2", true)
+        , inport3_("inport3", true)
+        , outport_("outport", false)
+        , alpha_("alpha", "Alpha", 1.0f, 0.0f, 1.0f, 0.001f)
+        , shader_("img_channel_combine.frag")
+    {
+        shader_.onReload([this]() { invalidate(InvalidationLevel::InvalidResources); });
+
+        addPort(inport0_);
+        addPort(inport1_);
+        addPort(inport2_);
+        inport3_.setOptional(true);
+        addPort(inport3_);
+
+        addPort(outport_);
+
+        addProperty(alpha_);
+    }
+
+    void ImageChannelCombine::process() {
+        if (inport0_.getData()->getDimensions() != inport1_.getData()->getDimensions() ||
+            inport0_.getData()->getDimensions() != inport2_.getData()->getDimensions() ||
+            inport3_.hasData() && inport0_.getData()->getDimensions() != inport3_.getData()->getDimensions()) {
+            LogError("inport dimensions don't match!");
+            return;
+        }
+
+        if (inport0_.isChanged() || inport1_.isChanged() || inport2_.isChanged() || inport3_.isChanged()) {
+            const auto dimensions = inport0_.getData()->getDimensions();
+            auto img = std::make_shared<Image>(dimensions, DataVec4UInt8::get());
```

Comment:
This is not good! Why is the image format hard coded here? Either use the common denominator of all input formats or at least use the one of the first inport.